### PR TITLE
feat(documents): fallback on text extension when pdf is not available

### LINKED ISSUE #110: Render documents for Toulouse
## Context

To give some context, this is how we currently store in FHIR for Toulouse the content of the documents available

```json
{
  "resourceType" : "DocumentReference",
  
  "extension": [ 
  
  {
  "url": "http://arkhn-orbis-thrombectomie.com",
  "valueString": "This is the first part of the document \r\r Operator: Dr John DOE  \r Etc Etc"
  }, 

  { 
  "url": "http://arkhn-orbis-thrombectomie.com",
  "valueString": "Second part of the same document.  \r\r Etc etc"
  }  ],

  "...": "...",

}
```

As far as I know this is not very standard FHIR-ly speaking, but was the quickest solution to implement as Cohort already checks this extension when searching on documents.

here
image

or here while creating a cohort
image

## Issue

Now that we can search on those documents, we want to be able to show them.

image

Today it seems that Cohort knows how to render a DocumentReference when it contains an external reference to an image (png for instance) or a PDF. 

```json
"content": [ {
    "attachment": {
          "url": "file://992327893739375.pdf",
          "title": "file://992327893739375.pdf"
     }
} ],
```

In our case in Toulouse, it doesn't render anything as the DocumentReference directly contains a string which is the text extracted from the original document.

I raise 2 questions here @elsiehoffet-94 @Peltier10 @simonvadee 

- In Toulouse, would it be possible to have access to the PDF documents, in order to render them the same way that we do today ?
- Does it make sense to have in Cohort a kind of "fallback strategy" when rendering a DocumentReference ? Something like : if I don't find any external reference but there is the full text extract, just render it on the fly as a basic PDF. We may encounter the same situation in other hospitals.

### PR DESCRIPTION
**Pull request by simonvadee**
## Fixes
Fixes https://github.com/arkhn/Cohort360/issues/110

screenshot coming soon

### CODE REVIEW

File: src/components/Cohort/Documents/DocumentList/DocumentList.tsx [hunk 1]
```diff
@@ -138,7 +145,7 @@ const DocumentRow: React.FC<DocumentRowTypes> = ({
         minute: '2-digit'
       })
     : ''
-
+  console.log(row, `${FILES_SERVER_URL}${row.content[0].attachment?.url?.replace(/^file:\/\//, '')}`)
```
Comment (tevariou) [inline src/components/Cohort/Documents/DocumentList/DocumentList.tsx]: rm?

File: src/components/Cohort/Documents/PdfRenderer/PdfRenderer.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+})
+
+type PdfRendererProps = {
+  content?: string
+}
+
+const PdfRenderer: React.FC<PdfRendererProps> = ({ content }) => {
+  if (!content) return <p></p>
+  const sections = content.split('\n\n').filter(Boolean)
+  console.log(sections)
```
Comment (tevariou) [inline src/components/Cohort/Documents/PdfRenderer/PdfRenderer.tsx]: rm?

File: src/components/Cohort/Documents/PdfRenderer/PdfRenderer.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    fontSize: '1.3rem'
+  }
+})
+
+type PdfRendererProps = {
+  content?: string
+}
+
+const PdfRenderer: React.FC<PdfRendererProps> = ({ content }) => {
+  if (!content) return <p></p>
```
Comment (tevariou) [inline src/components/Cohort/Documents/PdfRenderer/PdfRenderer.tsx]: maybe make content required and check if there is one before calling PdfRenderer

**Review by tevariou — CHANGES_REQUESTED** [review]

**Comment by MiskoG** [comment]
Ping me when the screenshot is available 😊

**Comment by simonvadee** [comment]
Screenshot 2021-07-01 at 11 01 22

**Review by MiskoG — APPROVED** [review]
😍 Awesome !

File: src/components/Cohort/Documents/PdfRenderer/PdfRenderer.tsx (hunk 3 above)
Comment (simonvadee) [reply src/components/Cohort/Documents/PdfRenderer/PdfRenderer.tsx]: this is what I started to do in the other component but then I forgot to do it here 😓

**Review by tevariou — APPROVED** [review]
lgtm!

**Pull request merged**